Update Core.php

gstesq opened the pull request:
When exporting JS using module assets folder, server returning MIME type 'text/plain' - needs 'text/javascript'

gstesq reviewed (commented):
Comments?

jakecastelli commented:
Hi Gray, I will try to review this a bit later if that is ok.

jakecastelli reviewed (approved):
This is valid! If you do not set the Content-Type then the JavaScript Module (import/ export) will not work.

jakecastelli commented on engine/Core.php:
```diff
@@ -110,6 +114,9 @@ private function serve_child_module_asset($asset_path, $file_name) {
                         if (is_numeric($pos2)) {
                             $content_type = 'text/css';
                         }
+                        if (is_numeric($pos2)) {
```
don't you need this line here `$pos2 = strpos($file_name, '.js');` ?

gstesq replied:
Yeah - thought I'd added it. Cheers.

gstesq replied:
Also, I had to add another check: if ($content_type == 'text/html') { $pos2 = strpos($file_name, '.js'); if (is_numeric($pos2)) { $content_type = 'text/javascript'; } }

When using string literals in the javascript it was returning html.

gstesq commented:
Also, I had to add another check:`                    if ($content_type == 'text/html') {
                        $pos2 = strpos($file_name, '.js');
                        if (is_numeric($pos2)) {
                            $content_type = 'text/javascript';
                        }
                    }`

gstesq commented on engine/Core.php:
```diff
@@ -105,12 +117,21 @@ private function serve_child_module_asset($asset_path, $file_name) {
+                        }
+                    }
+                    
+                    if ($content_type == 'text/html') {
```
Added to return 'text/javascript' instead of 'text/html' when using string literals.

jakecastelli commented:
Just a small suggestion Gary, you can condense them by saying `if ($content_type == 'text/plain' || $content_type == 'text/html')`

gstesq commented:
> Just a small suggestion Gary, you can condense them by saying `if ($content_type == 'text/plain' || $content_type == 'text/html')`

Refactored and updated.

davidjconnelly reviewed (approved):
Great and thank you!  If there's anything else, please do tweak away and submit another pull request.  I'm currently uploading to a new host and my brain is functioning at about 1%.  So, just do what you need to do.  

Super grateful, as always.  Thanks!
